### ISSUE TITLE
Wrong number of arguments (given 1, expected 0)

### ISSUE THREAD

**Issue by saurabhshinde02** [exAspArk/batch-loader#73]
Hi,
I tried the example given here https://github.com/exAspArk/batch-loader#batch-key after updating ruby version to 3.0.0.

```
  id = post.association_id
  key = post.association_type

  BatchLoader.for(id).batch(key: key) do |ids, loader, args|
    model = Object.const_get(args[:key])
    model.where(id: ids).each { |record| loader.call(record.id, record) }
  end
```

It gives me `wrong number of arguments (given 1, expected 0)` error when passing arguments in `batch` method, Now, It is not expecting any argument. Is there any alternative way to do the same?

ruby version: 3.0.0
rails version: 6.0.3.4
graphql version: (>= 1.8)

**Comment by an-lee** [comment]
Same problem here. Merge this PR https://github.com/exAspArk/batch-loader/pull/71 should fix this.

**Comment by saurabhshinde02** [comment]
Hi, any update on the issue?

**Comment by adammeghji** [comment]
+1 would be great to see batch-loader be compatible with ruby 3.0.0! 😍

**Comment by jandudulski** [comment]
@exAspArk any chances to take a look? :pray:

**Comment by ThiagoAnunciacao** [comment]
🙏🏼  👍🏼

**Comment by exAspArk** [comment]
Should be fixed by #71. Thanks @an-lee! 🎉 

I released the new `2.0.0` version. Here is the [changelog](https://github.com/exAspArk/batch-loader/blob/master/CHANGELOG.md#v200). Please let me know if there are any issues 🙏
